Sources/Controllers/Map/Widgets/OASunriseSunsetWidget.mm

```diff
@@ -369,4 +409,33 @@ - (NSString *) formatNextTime:(NSTimeInterval)nextTime
     return [dateFormatter stringFromDate:nextDate];
 }
 
+- (void) updateCachedLocation
+{
+    CLLocation *newCenterLatLon = [[OARootViewController instance].mapPanel.mapViewController getMapLocation];
```

Comment:
it seems your can use CLLocationCoordinate2D loc = [OAResourcesUIHelper getMapLocation]; ?